### PR TITLE
Fix using event configurator with DI references

### PR DESCRIPTION
**Pull request by viktorprogger**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | ❌

Fixing usage with aliases in DI configuration:
`['someAlias' => ClassName::class]` in DI configuration and `['someALias', 'methodName']` in `events-console.php`.

### CODE REVIEW

File: src/Config/EventConfigurator.php [hunk 1]
```diff
@@ -54,6 +54,14 @@ private function isCallable($definition): bool
             return true;
         }
 
-        return is_array($definition) && array_keys($definition) === [0, 1] && in_array($definition[1], get_class_methods($definition[0]) ?? [], true);
+        return is_array($definition)
+            && array_keys($definition) === [0, 1]
+            && (
+                in_array($definition[1], get_class_methods($definition[0]) ?? [], true)
```
Comment (roxblnfk) [inline src/Config/EventConfigurator.php:60]: Here we must be sure that the `$definition[0]` is string (class name) or object

File: src/Config/EventConfigurator.php [hunk 2]
```diff
@@ -54,6 +54,14 @@ private function isCallable($definition): bool
...
+                || (
+                    $this->container->has($definition[0])
+                    && method_exists($this->container->get($definition[0]), $definition[1])
```
Comment (roxblnfk) [inline src/Config/EventConfigurator.php:63]: Here we must be sure that the $definition[0] is string

**Comment by viktorprogger** [comment]
The same is made in https://github.com/yiisoft/yii-web/pull/282, but with test modifing.

**Comment by viktorprogger** [comment]
> Now BC is broken
> Because `$definition[0]` can also be an `object`

@roxblnfk, `is_callable` returns true when `$definition[0]` is an object and `$definition[1]` is a correct method name.

**Comment by samdark** [comment]
Closing since we're going to move configurator into yiisoft/yii-event anyway.

**Pull request closed without merging**